src/main/java/com/dawkastresu/medical_clinic/PatientService.java

```diff
@@ -0,0 +1,29 @@
+package com.dawkastresu.medical_clinic;
+
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@RequiredArgsConstructor
+@Service
+public class PatientService {
+    private final PatientRepository patientRepository;
+
+    public void editPatientByEmail(String email, Patient newPatient) {
+        Patient patient = patientRepository.findPatientByEmail(email)
+                .orElseThrow(IllegalArgumentException::new);
```

Comment:
Przekazujmy message do wyjatkow w tym przypadku:
() -> new IllegalArgumentException("Patient not found")